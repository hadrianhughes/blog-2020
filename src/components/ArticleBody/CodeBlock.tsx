import React, { FunctionComponent } from 'react'
import SyntaxHighlighter from 'react-syntax-highlighter/dist/cjs/prism'
import solarizedlight from 'react-syntax-highlighter/dist/cjs/styles/prism/solarizedlight'
import materialDark from 'react-syntax-highlighter/dist/cjs/styles/prism/material-dark'

interface CodeBlockProps {
  value: string;
  language: string;
  darkMode?: boolean;
}

const CodeBlock: FunctionComponent<CodeBlockProps> = ({ value, language, darkMode }) => (
  <SyntaxHighlighter language={language} style={darkMode ? materialDark : solarizedlight}>
    {value || ''}
  </SyntaxHighlighter>
)

export default CodeBlock
