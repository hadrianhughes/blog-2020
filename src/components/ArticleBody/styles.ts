import styled from 'styled-components'
import { rem, minWidth } from '../../styles/tools'

export const Container = styled.section`
  padding: 0 ${rem(10)};
`

export const Heading = styled.h1`
  font-size: 2rem;

  ${minWidth('medium')} {
    font-size: 3rem;
  }
`

export const Paragraph = styled.p`
  text-align: justify;
`
