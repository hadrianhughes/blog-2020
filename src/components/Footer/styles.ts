import styled from 'styled-components'

export const StyledFooter = styled.footer`
  align-items: flex-end;
  display: flex;
  font-family: Karla, sans-serif;
  justify-content: center;
  grid-column: 1 / -1;
`
