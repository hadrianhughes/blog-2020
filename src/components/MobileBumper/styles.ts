import styled from 'styled-components'

export const Container = styled.section`
  display: flex;
  justify-content: space-between;
`

export const Heading = styled.h1`
  margin: 0;
`
