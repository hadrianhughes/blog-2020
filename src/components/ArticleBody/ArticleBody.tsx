import React, { FunctionComponent } from 'react'
import { Document, BLOCKS } from '@contentful/rich-text-types'
import { documentToReactComponents } from '@contentful/rich-text-react-renderer'
import { Container, Heading, Paragraph } from './styles'

interface ArticleBodyProps {
  heading: string
  data: Document
}

const options = {
  renderNode: {
    [BLOCKS.PARAGRAPH]: (_, children) => <Paragraph>{children}</Paragraph>
  }
}

const ArticleBody: FunctionComponent<ArticleBodyProps> = ({ heading, data }) => (
  <Container>
    <Heading>{heading}</Heading>
    {documentToReactComponents(data, options)}
  </Container>
)

export default ArticleBody
