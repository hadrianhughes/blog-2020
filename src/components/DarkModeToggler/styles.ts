import styled, { css } from 'styled-components'
import Text from '../Text'
import { colors, spacings } from '../../styles/settings'
import { rem, minWidth } from '../../styles/tools'

const toggleWidth = 50
const toggleHeight = 30
const sliderSize = toggleHeight - 8

export const Label = styled(Text)`
  ${({ $darkMode }): string => css`
    align-items: center;
    color: ${$darkMode ? colors.white : colors.grey};
    display: flex;
    font-family: Karla, sans-serif;
    justify-content: center;
    margin-top: ${rem(spacings.large)};
    transition: color 0.2s;

    ${minWidth('medium')} {
      margin-top: ${rem(spacings.xlarge)};
    }

    span {
      line-height: 1;
      max-width: ${rem(50)};
    }
  `}
`

export const Checkbox = styled.input`
  appearance: none;
  background-color: ${colors.nightGrey};
  border: 2px solid ${colors.nightGrey};
  border-radius: ${rem(toggleHeight)};
  cursor: pointer;
  height: ${rem(toggleHeight)};
  margin: 0;
  margin-left: ${rem(spacings.small)};
  position: relative;
  width: ${rem(toggleWidth)};

  &::after {
    background-image: url(/img/sun.svg);
    background-size: contain;
    content: '';
    position: absolute;
    height: ${rem(sliderSize)};
    left: 2px;
    top: 50%;
    transform: translateY(-50%);
    transition: transform 0.2s;
    width: ${rem(sliderSize)};
  }

  &:checked {
    background-color: ${colors.lightGrey};
    border: 2px solid ${colors.lightGrey};

    &::after {
      background-image: url(/img/moon.svg);
      transform: translate(calc(100% - 2px), -50%);
    }
  }
`
