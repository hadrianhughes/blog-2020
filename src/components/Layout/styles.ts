import styled from 'styled-components'
import { minWidth, rem } from '../../styles/tools'
import { spacings } from '../../styles/settings'

export const Container = styled.div`
  ${minWidth('large')} {
    display: grid;
    grid-template-columns: 1fr 2.5fr;
    min-height: 100vh;
  }
`

export const Header = styled.header`
  div {
    padding: ${props => props.isBigScreen ? rem(spacings.medium) : rem(spacings.small)};
    position: sticky;
    top: 0;
  }
`

export const Main = styled.main`
  padding: ${rem(spacings.medium)};
`
