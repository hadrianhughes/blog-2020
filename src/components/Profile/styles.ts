import styled from 'styled-components'
import { minWidth, rem } from '../../styles/tools'
import { spacings, colors } from '../../styles/settings'

export const ProfileImg = styled.img`
  border-radius: 50%;
  display: block;
  margin: 0 auto;
  width: 60%;
`

export const Container = styled.section`
  margin-top: ${rem(spacings.large)};

  ${minWidth('small')} {
    margin-top: 0;
  }
`
