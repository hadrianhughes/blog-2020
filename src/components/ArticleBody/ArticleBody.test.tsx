import React from 'react'
import { mount } from 'enzyme'
import { Document, BLOCKS } from '@contentful/rich-text-types'
import toJson from 'enzyme-to-json'
import ArticleBody from './ArticleBody'
import { Container, Heading } from './styles'

const data: Document = {
  data: {},
  nodeType: 'document' as BLOCKS.DOCUMENT,
  content: [
    {
      data: {},
      nodeType: 'paragraph' as BLOCKS.PARAGRAPH,
      content: [
        {
          data: {},
          nodeType: 'text',
          marks: [],
          value: 'Hello World'
        }
      ]
    }
  ]
}

describe('ArticleBody component', () => {
  const heading = 'test heading'
  const wrapper = mount(<ArticleBody data={data} heading={heading} />)

  it('Should render a <Container>', () => {
    expect(wrapper.find(Container).exists()).toBeTruthy()
  })

  it('Should render the `heading` prop in a <Heading>', () => {
    expect(wrapper.find(Heading).exists()).toBeTruthy()
    expect(wrapper.find(Heading).text()).toBe(heading)
  })

  it('Should pass the `data` prop to the `documentToReactComponents` function', () => {
    expect(toJson(wrapper.find('section').children())).toMatchSnapshot()
  })
})
