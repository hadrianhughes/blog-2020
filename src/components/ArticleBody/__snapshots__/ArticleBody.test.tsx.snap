// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleBody component Should pass the \`data\` prop to the \`documentToReactComponents\` function 1`] = `
Array [
  <styled.h1>
    <h1
      className="sc-gtssRu hiTVzU"
    >
      test heading
    </h1>
  </styled.h1>,
  <styled.p
    key="0"
  >
    <p
      className="sc-dlnjPT dkPOOf"
    >
      Hello World
    </p>
  </styled.p>,
]
`;
